include/fluent-bit/flb_aws_util.h

```diff
@@ -177,5 +178,12 @@ int flb_read_file(const char *path, char **out_buf, size_t *out_size);
 flb_sds_t flb_get_s3_key(const char *format, time_t time, const char *tag,
                          char *tag_delimiter, uint64_t seq_index);
 
+/*
+ * Replace %f and %L in time_key_format with milliseconds and nanoseconds. And get the
+ * time format output and output length with strftime().
+ */
+size_t create_time_formatter(char **out_buf, const char *time_key_format,
```

Comment:
nit: this does not create time formatters, and also I think we need to prefix it with FLB and may be aws too like other functions in here: I would call it: `flb_aws_strftime_nanos` or `flb_aws_strftime_precision`... since this is an extension to strftime basically.